src/backend/distributed/shared_library_init.c

```diff
@@ -606,12 +606,20 @@ RegisterCitusConfigVariables(void)
 
 	DefineCustomBoolVariable(
 		"citus.subquery_pushdown",
```

Comment:
I wonder if it would make sense to have the name as `unsafe_subquery_pushdown` or `push_subquery_without_checks` ?

Reply:
That requires the users/scripts needs to be updated, so we prefer those changes with major version upgrades (such as you keeping many task-tracker GUCs in the code). When we do the next major version upgrade, I'll keep this in mind.

Reply:
Yeah I was asking in general, since that could clearly and explicitly convey the message that it is not `safe` to use. Otherwise people will probably not read the description and when they see `subquery_pushdown` they will think `why not` :)

Reply:
Instead of trying to remember, lets either make Issues for these, and mark them Citus10 or sth. We can go through that list once we decide to major bump. I feel like if we rely on eng to remember what we want to change in a major version change it is too easy to forget.

For a guc like this we probably could also have them side by side working no the same var. We can add a deprecation warning to this GUC so that when a user sets them they already see a warning popping up but the change is still successfully applied.

Reply:
Marking this as related https://github.com/citusdata/citus/issues/2146, I think we could keep adding to that issue (or something else)

> We can add a deprecation warning to this GUC so that when a user sets them they already see a warning popping up but the change is still successfully applied.

makes sense, I'll do that. We already do something similar when you set real-time executor, where we ignore it. This time, we'll warn the user but still use the value